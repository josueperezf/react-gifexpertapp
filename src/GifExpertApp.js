import { Fragment, useState } from "react";
import { AgregarCategoria } from "./componentes/AgregarCategoria";
import GifGrid from "./componentes/GifGrid";
import 'animate.css'
const GifExpertApp = ()=>{
    const [categorias, setcategorias] = useState (['One Punch']);

    return (
        <Fragment>
            <h2>GifExpertApp</h2>
            <AgregarCategoria setcategorias={setcategorias} />
            <hr/>
            
            <ol>
                {
                    categorias.map((categoria) => <GifGrid key={categoria} categoria={categoria}  /> )
                }
            </ol>

        </Fragment>
    );
}

export default GifExpertApp;