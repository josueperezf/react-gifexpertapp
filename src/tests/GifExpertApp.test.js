import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe('Pruebas en <GifExpertApp/>',()=>{

    test('debe mostrarse correctamente', () => {
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
    });
    
    test('debe de mostrar una lista de categorias', () => {
        const categorias = ['One Punch', 'goku'];
        const wrapper = shallow(<GifExpertApp defaultCategoria={categorias} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categorias.length);
    })
    
});