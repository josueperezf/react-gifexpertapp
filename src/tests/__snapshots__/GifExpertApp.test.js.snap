// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp/> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AgregarCategoria
    setcategorias={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      categoria="One Punch"
      key="One Punch"
    />
    <GifGrid
      categoria="goku"
      key="goku"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GifExpertApp/> debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AgregarCategoria
    setcategorias={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
